src/plugins/conditionalStyles/ConditionalStylesInspectorViewProvider.js

```diff
@@ -0,0 +1,80 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2020, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+import ConditionalStylesView from './components/ConditionalStylesView.vue'
+import Vue from 'vue';
+
+export default class ConditionalStylesInspectorViewProvider {
+    /*
+    * @param supportedObjectTypesArray is a list of types that can show conditional styles in the inspector view
+    */
+    constructor(openmct, supportedObjectTypesArray) {
+        this.openmct = openmct;
+        this.key = 'conditional-styles-inspector';
+        this.name = 'Style';
+        this.supportedObjectTypesArray = supportedObjectTypesArray;
+    }
+
+    canView(selection) {
+        if (selection.length === 0 || selection[0].length === 0) {
+            return false;
+        }
+        let object = selection[0][0].context.item;
+
+        return object && this.supportedObjectTypesArray.some(type => object.type === type);
+    }
+
+    view(selection) {
+        let component;
+        let openmct = this.openmct;
+        let name = this.name;
+        return {
+            show: function (element) {
+                component = new Vue({
+                    provide: {
+                        openmct
+                    },
+                    el: element,
+                    components: {
+                        ConditionalStylesView
+                    },
+                    template: '<conditional-styles-view></conditional-styles-view>'
+                });
+            },
+            destroy: function () {
+                if (component) {
+                    component.$destroy();
+                    component = undefined;
+                }
+            },
+            tabbed() {
```

Comment:
Note: This deviates from the API - need a suggestion to move the tabbed and name methods into the provider instead of the view property.

Reply:
After some reflection, I think that the right approach here is to add a new style inspector view registry. This would need to be added to the Open MCT API.

It would be called like so
``` JavaScript
openmct.styleInspector.addViewProvider(new ConditionalStylesInspectorViewProvider(...));
```

It would just be a regular ViewRegistry object, like the object view registry and the existing inspector view registry.

If we're doing this we should also refactor `openmct.inspectorViews` to `openmct.propertiesInspector` to make it clear which tab it is referencing.

Note that I have dropped the `views` from `openmct.*InspectorViews` and renamed `ViewRegistry.addProvider` to `ViewRegistry.addViewProvider`. This is because `openmct.propertiesInspectorViews` is just too long a property name for a public API :)

This will require changes in VISTA, which defines some inspector views. It will just be a global find and replace though, should be fairly straightforward. Just FYI @deeptailor

Reply:
Sounds good. See commits:
https://github.com/nasa/openmct/pull/2655/commits/bf2aa8483971a926ea85ac4bf0c9587d3e44a1ee
and

Reply:
I should have thought about this some more in the context of the existing conditionals design. Sorry about this.

Honestly, the style inspector doesn't need to be a registry at all right now. It's not going to have sections, it's not something that we want to provide API for extending right now, so we should keep it as simple as possible. It can be as simple as a Vue component, hard coded into the inspector.

As such, we should probably rename the `.propertiesInspector` back to just the `.inspector` as well *ducks*. Let's keep things simple for now.

Reply:
ok will do